packages/create-app/src/index.ts

```diff
@@ -0,0 +1,410 @@
+            if (overwrite === 'no') {
+              throw new Error(`${red('✖')} Operation cancelled`)
+            }
+            return null
+          },
+          name: 'overwriteChecker',
+        },
+      ],
+      {
+        onCancel: () => {
+          throw new Error(`${red('✖')} Operation cancelled`)
+        },
+      },
+    )
+  } catch (error: unknown) {
+    if (error instanceof Error) {
+      console.log(error.message)
+    }
+    return
+  }
+
+  // User choices associated with prompts
+  const { rootDir, templates, overwrite } = result
+
+  const rootPath = path.join(cwd, rootDir)
+  const pkgInfo = pkgFromUserAgent(process.env.npm_config_user_agent)
+  const pkgManager = pkgInfo ? pkgInfo.name : 'npm'
+
+  // Step 2. Copy the templates
+
+  console.log(`\nScaffolding templates in ${rootPath}...`)
+
+  for (const template of templates) {
+    const callerPath = fileURLToPath(import.meta.url)
+    const templateDir = `template-${template.name}`
+    const templateSrc = path.resolve(callerPath, '../..', templateDir)
+    const templateDest = path.join(rootPath, template.targetDir ?? `packages/${template.name}`)
+
+    // Exit if no template files found
+    if (!fs.existsSync(templateSrc)) {
+      continue
+    }
+
+    if (fs.existsSync(templateDest) && overwrite === 'yes') {
+      emptyDir(templateDest)
+    } else {
+      fs.mkdirSync(templateDest, { recursive: true })
+    }
+
+    const write = (file: string, content?: string) => {
+      const targetPath = path.join(templateDest, moveFiles[file] ?? file)
+      if (content) {
+        fs.writeFileSync(targetPath, content)
+      } else {
+        copy(path.join(templateSrc, file), targetPath)
+      }
+    }
+
+    const files = fs.readdirSync(templateSrc)
+    for (const file of files) {
+      write(file)
+    }
+  }
+
+  // Step 3. Run the custom commands
+
+  console.log(`\nDone. Now running template commands.`)
+
+  // Flatten all selected template commands
+  const customCommands = templates
+    .map((t) => t.customCommands)
+    .flat()
+    .filter(Boolean) as Command[]
+
+  // Run commands from project root
+  if (rootPath !== cwd) {
+    spawnSync('cd', [rootPath], { stdio: 'inherit' })
+  }
+
+  for (const customCommand of customCommands) {
+    // `node:fs` commands as are stricly typed arrays
+    if (Array.isArray(customCommand)) {
+      const [command, ...args] = customCommand
+      const formattedArgs = args.map((arg) =>
+        typeof arg === 'string' ? formatCommand(arg) : arg,
+      ) as typeof args
+      fs[command](...formattedArgs)
+      continue
+    }
+
+    // `spawnSync` commands are loosely typed strings
+    const formattedCommand = formatCommand(customCommand)
+    const [command, ...args] = splitCommand(formattedCommand) as [string, ...string[]]
+    spawnSync(command, args, { stdio: 'inherit' })
+  }
+
+  // Step 4. Install dependencies
+
+  console.log(`\nDone. Now installing dependencies.\n`)
+
+  if (pkgManager) {
```

Comment:
**chore:** Make the `pkgManager` type `string` and not `string | undefined`

Reply:
Updated :)